qualtran/bloqs/qsp/fast_qsp.py

```diff
@@ -0,0 +1,84 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import numpy as np
+from scipy.optimize import minimize
+
+
+class FastQSP:
+    def __init__(self, poly):
+        self.conv_p_negative = self.complex_conv_by_flip_conj(poly.real, poly.imag) * -1
+        self.conv_p_negative[poly.shape[0] - 1] = 1 - np.linalg.norm(poly) ** 2
+
+    def normalize(self, input_poly, granularity=8):
+        P = np.pad(input_poly, (0, 2**granularity - input_poly.shape[0]))
+        ft = np.fft.fft(P)
+
+        # Normalize P
+        P_norms = np.abs(ft)
+        return input_poly / np.max(P_norms)
```

Comment:
why are we dividing by the maximum here? (in case it's explained in the paper, add a comment linking to the para/equation etc. in the paper)